source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -900,6 +901,293 @@ void MSETRequest::onChildResponse(Common::Redis::RespValuePtr&& value, uint32_t
+  }
+
+  if (!pending_request.handle_) {
+    pending_request.onResponse(Common::Redis::Utility::makeError(Response::get().NoUpstreamHost));
+    return nullptr;
+  }
+
+  return request_ptr;
+}
+
+void ScanRequest::onChildResponse(Common::Redis::RespValuePtr&& value, int32_t index, int32_t shard_index) {
+  // Request handled successfully
+  pending_requests_[index].handle_ = nullptr;
+  // Moving the response to pending response for doing validation later
+  // Incrementing the number of pending responses, it will drained during the validation
+  int64_t count = 2;
+
+  // Checking the cursor and number of objects for child request
+  std::string cursor = value->asArray()[0].asString();
+  int64_t objects = value->asArray()[1].asArray().size();
+  std::string newCount = std::to_string(count - objects);
+
+  // Resizing pending repsonses array based on the incoming reponses
+  if (index >= static_cast<int32_t>(pending_responses_.size())) {
+        // Resize the vector to accommodate the new index
+        pending_responses_.resize(index + 1);
+  }
+  ENVOY_LOG(debug,"response recived for index: '{}'", shard_index);
+
+  pending_responses_[index] = std::move(value);
+  ++num_pending_responses_;
+  bool send_response = true;
+
+  // Following conditions needs to be satisfied for making a child request
+  // 1) If cursor is zero, objects less than count
+  //  1.1) If No more shards to scan, there won't be any child request
+  //  1.2) If shards present, increment the shard index, update the count value, set cursor to 0 and send child request
+  // 2) If cursor is not zero
+  //  2.1) If objects returned is equal to count, there won't be any child request
+
+  if (cursor == "0" && objects < count && shard_index+1 < num_of_Shards_) {
+    // Popping the pending request, not needed anymore
+    pending_requests_.pop_back();
+    send_response = false;
+    // Cursor is always zero for child request
+    // Create the child request based on the original request
+    Common::Redis::RespValue child_request = request_;
+
+    // Set the cursor to "0" in the child request
+    child_request.asArray()[1].asString() = "0";
+    // Setting the new count for the child request
+    for (size_t i = 2; i < child_request.asArray().size() - 1; ++i) {
+      if (child_request.asArray()[i].asString() == "count") {
+          child_request.asArray()[i + 1].asString() = newCount;
+          break; // Stop searching after updating count
+      }
+    }
+    ENVOY_LOG(debug, "Child request: {}", request_.toString());
+    PendingRequest& pending_request = pending_requests_.back();
+    pending_request.handle_= makeScanRequest(route_, shard_index+1, child_request, pending_request, callbacks_.transaction());
+    if (!pending_request.handle_) {
+      pending_request.onResponse(Common::Redis::Utility::makeError(Response::get().NoUpstreamHost));
```

Comment:
why recursively call response here , just send the response from here

Reply:
Fixed this